vpr/src/route/connection_box_lookahead_map.h

```diff
@@ -2,16 +2,88 @@
 #define CONNECTION_BOX_LOOKAHEAD_H_
 
 #include <vector>
+#include <list>
 #include "physical_types.h"
 #include "router_lookahead.h"
+#include "router_lookahead_map_utils.h"
+#include "connection_box.h"
+
+template<typename T>
+class BoundingBox {
```

Comment:
This should probably be in its own header, and have unit tests.  Consider putting it in https://github.com/SymbiFlow/vtr-verilog-to-routing/tree/master%2Bwip/libs/libvtrutil/src ?

Reply:
Agreed.

Reply:
Agreed. I found `vtr::Rect`, so I'm adding to that instead. Note that `vtr::RectUnion` is not what I want since it is an exact union.